test_vm/src/lib.rs

```diff
@@ -71,10 +72,126 @@ use std::collections::HashMap;
 use std::error::Error;
 use std::fmt;
 use std::ops::Add;
+use util::apply_ok_;
+use util::pk_addrs_from;
 
+pub mod bench;
+pub mod executor;
 pub mod util;
 
-pub struct VM<'bs> {
+pub trait VM {
+    fn send_message(
+        &self,
+        from: Address,
+        to: Address,
+        value: TokenAmount,
+        method: MethodNum,
+        params: Option<IpldBlock>,
+    ) -> Result<MessageResult, TestVMError>;
+
+    fn resolve_address(&self, addr: &Address) -> Option<Address>;
+    fn create_accounts_seeded(
```

Comment:
I don't think this method should be on the trait, only the most fundamental operations. This is just a send and an address resolution. Even address resolution I'm not sure of, because it's just a state inspection and then code that could run outside the VM.